### ISSUE TITLE
Version 4.5 requires Python 3.6?

### ISSUE THREAD

**Issue by banjophil** [instaloader/instaloader#765]
After upgrading to 4.5 on a raspberry pi getting this error:

```
Traceback (most recent call last):
  File "/usr/local/bin/instaloader", line 5, in <module>
    from instaloader.__main__ import main
  File "/usr/local/lib/python3.5/dist-packages/instaloader/__init__.py", line 16, in <module>
    from .instaloader import Instaloader
  File "/usr/local/lib/python3.5/dist-packages/instaloader/instaloader.py", line 16, in <module>
    from typing import Any, Callable, ContextManager, IO, Iterator, List, Optional, Set, Union, cast
ImportError: cannot import name 'ContextManager'
```

**Comment by aandergr** [comment]
Which version of Python do you use? According to the Python documentation, [typing.ContextManager](https://docs.python.org/3/library/typing.html#typing.ContextManager) is supported in Python >= 3.5.4.

**Comment by banjophil** [comment]
3.5.3

**Comment by aandergr** [comment]
Okay, then I'd recommend to update Python to at least a more recent 3.5 release. If you have to stick to 3.5.3 for some reason, you could remove the occurrences of `ContextManager` (with that capitalization) from `instaloader/instaloader.py`, since it is used for type checking only.

**Comment by banjophil** [comment]
Sure, I'm also happy using an older Instaloader release for the moment. I thought we should maybe change the documentation to mention the new version only supports Python >= 3.5.4. I'm also new to github, so sorry if I'm not doing this right.

**Comment by aandergr** [comment]
> I thought we should maybe change the documentation to mention the new version only supports Python >= 3.5.4.

As for now, given that the limitation to >= 3.5.4 stems from an optional type cast, I'd prefer that we change the code such that it just works on Python 3.5.3. I introduced that incompatibility with updating the version of MyPy, which we use for typechecking in our CI pipeline, and thereby updating our code to make use of more solid typing features, which are also available on recent Python 3.5 releases (2a3711687fe4c163d8ccebf6a900a8198cb71493).

According to `the-psf:pypi.downloads` BigQuery table, only 0.6 percent of Instaloader 4.5 installations have been done with Python 3.5, so a step in the not-so-far future might be to drop support for Python 3.5 entirely in one of the next bigger releases and get rid of the limitations of supporting such an old Python version. Python 3.6 has been released more than three years ago, and only few people seem to use recent software on an outdated operating system. Dropping Python 3.5 support would give us at least literal string formatting, the file system path protocol, variable annotations, guaranteed availability of BLAKE2 hash and better unicode support on Windows.
